src/screens/EditProfileScreen.tsx

```diff
@@ -0,0 +1,135 @@
+import { StyleSheet, Text, TouchableOpacity } from 'react-native'
+import RNPickerSelect from 'react-native-picker-select'
+import React from 'reactn'
+import { ActivityIndicator, Divider, TextInput, View } from '../components'
+import { PV } from '../resources'
+import { getAuthUserInfo, updateLoggedInUser } from '../state/actions/auth'
+import { core, navHeader } from '../styles'
+
+type Props = {
+  navigation?: any
+}
+
+type State = {
+  isLoading?: boolean
+  name?: string
+  selectedIsPublicKey?: string
+}
+
+export class EditProfileScreen extends React.Component<Props, State> {
+
+  static navigationOptions = ({ navigation }) => {
+    return {
+      title: 'Edit Profile',
+      headerRight: (
+        <TouchableOpacity
+          onPress={navigation.getParam('updateUser')}>
+          <Text style={navHeader.textButton}>Save</Text>
+        </TouchableOpacity>
+      )
+    }
+  }
+
+  constructor(props: Props) {
+    super(props)
+    const user = props.navigation.getParam('user')
+
+    this.state = {
+      isLoading: true,
+      name: user.name,
+      selectedIsPublicKey: user.isPublic
+    }
+  }
+
+  async componentDidMount() {
+    this.props.navigation.setParams({ updateUser: this._updateUser })
+    await getAuthUserInfo()
+    this.setState({ isLoading: false })
+  }
+
+  _updateUser = async () => {
+    this.setState({
+      isLoading: true
+    }, async () => {
+      const { name, selectedIsPublicKey } = this.state
+      const { id } = this.global.session.userInfo
+      await updateLoggedInUser({
+        id,
+        isPublic: selectedIsPublicKey,
+        name
+      }, this.global)
+      this.setState({ isLoading: false })
+      this.props.navigation.goBack(null)
+    })
+  }
+
+  _ItemSeparatorComponent = () => {
+    return <Divider />
+  }
+
+  _onChangeIsPublic = (key: string) => {
+    this.setState({ selectedIsPublicKey: key })
+  }
+
+  _onChangeName = (text: string) => {
+    this.setState({ name: text })
+  }
+
+  render() {
+    const { globalTheme } = this.global
+    const { isLoading, name, selectedIsPublicKey } = this.state
+    const selectedIsPublicOption = isPublicOptions.filter((x) => x.value === selectedIsPublicKey)[0] || selectPlaceholder
```

Comment:
if you use `.find()` instead of`.filter()`, a) the loop stops as soon as it finds the element so it doesn't go through the whole array, and b) you dont need to do the `[0]` part

Reply:
Ohh much better